common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/OperationCreator.java

```diff
@@ -130,36 +132,116 @@ private void checkWrappedTypeKotlinNullability(MethodInfo methodInfo,
                     .filter(f -> compareParameterLists(f.getValueParameters(), methodInfo.parameterTypes()))
                     .findAny();
             if (function.isPresent()) {
-                KmType returnType = function.get().getReturnType();
-                KmTypeProjection arg = returnType.getArguments().get(0);
-                int flags = arg.getType().getFlags();
-                boolean nullable = Flag.Type.IS_NULLABLE.invoke(flags);
-                if (nullable) {
-                    operation.setNotNull(false);
+                // handle return type
+                if (operation.hasWrapper()) {
+                    KmType returnType = function.get().getReturnType();
+                    var nullable = isKotlinWrappedTypeNullable(returnType);
+                    // TODO: only for CollectionOrArrayOrMap applicable? What about other Wrapper?
```

Comment:
I think the only other wrappers are reactive types, and that seems to be handled by the `else` branch, so I think it's ok